core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -308,27 +308,29 @@ class Partition(val topicPartition: TopicPartition,
                 s"different from the requested log dir $logDir")
             false
           case None =>
-            createLogIfNotExists(isNew = false, isFutureReplica = true, highWatermarkCheckpoints)
+            // not sure if topic ID should be none here, but not sure if we have access in ReplicaManager where this is called.
```

Comment:
TODO: remove this when we decide if we want to pass in None or a topicID

Reply:
I cannot think of a reason not to pass the topic id if we have it available. Otherwise, we would need logic to set it again after the log dir is swapped.